[Hub Generated] Publish private branch 'raghumunukutla-supportrp-Microsoft.Support-2023-06-01-preview'

raghumunukutla opened the pull request:
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://openapihub.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/published%2FAzure%2Fazure-rest-api-specs-pr%2Fraghumunukutla-supportrp-Microsoft.Support-2023-06-01-preview...main/).
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

> [!NOTE]
> As of January 2024 there is no PR assignee. This is expected. See https://aka.ms/azsdk/pr-arm-review.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![diagram](https://github.com/Azure/azure-rest-api-specs/assets/20296335/35140fb9-639e-4e44-ac4b-38b1a20dddf9)

<details>
<summary> Click here to see the details of Step 1 </summary>

### Breaking changes review (Diagram Step 1)

If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
- The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
- You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

</details>

<details> 
<summary> Click here to see the details of Step 2 </summary>

### ARM API changes review (Diagram Step 2)

- For help with Step 2 / "ARM review", see https://aka.ms/azsdk/pr-arm-review.

</details>

<details>
<summary> Click here to see the diagram footnotes</summary>

### Diagram footnotes

[1] See [ARM review queue] (for **PR merge** queues, see [2]).  
[2] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])  
The ARM reviewer on-call engineer visits the merge queue twice a day, so the approximate ETA for merges is 12 - 24 hours.

</details>

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [x] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been generated using [OpenAPI Hub](https://aka.ms/openapihub)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with PR workflow diagram Step 2 (ARM review), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://aka.ms/azsdk/pr-arm-review#about-the-arm-review-queue
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

raghumunukutla committed:
Adds base for updating Microsoft.Support from version preview/2022-09-01-preview to version 2023-06-01-preview

raghumunukutla committed:
Updates readme

raghumunukutla committed:
Updates API version in new specs and examples

vimunuku committed:
Changes for updating swagger and corresponding examples

vimunuku committed:
Added one example

vimunuku committed:
Changes to address check failures

vimunuku committed:
Changes to fix breaking changes

vimunuku committed:
Changes to address couple of other checks

vimunuku committed:
Minor change

vimunuku committed:
Minor fix

vimunuku committed:
Changes as per PR comments

vimunuku committed:
Changes for adding support to classifyProblems at tenant scope

vimunuku committed:
Minor change

vimunuku committed:
Changes for addressing comments

vimunuku committed:
1. Changes for adding classifyservices apis for both tenant and subscription scope
2. Changes for adding a new property related service for problem classification api

vimunuku committed:
Minor changes to fix failures related to automated checks

vimunuku committed:
Changes as per review comments

vimunuku committed:
Minor changes for updating the examples

vimunuku committed:
Minor change for updating examples

vimunuku committed:
Changes as per PR comments

vimunuku committed:
Minor change

vimunuku committed:
Minro change to example file

vimunuku committed:
Changes for adding: "x-ms-parameter-location" to parameters

vimunuku committed:
Changes for adding suppressions for additional properties

raghumunukutla commented:
The PR is created based on the updates in the private branch. The updates in the PR has already been reviewed and approved with this PR [Azure/azure-rest-api-specs-pr/15540](https://github.com/Azure/azure-rest-api-specs-pr/pull/15540)

vimunuku committed:
On top of signed off private preview PR, below are couple of changes added in this commit:

- Remove readonly for enrollmentId which is now also sent as create ticket payload
- Adding a new api(lookUpResourceId) which was in a different version of private preview(2019-05-01-preview)

vimunuku committed:
Minor change for fixing the model validation

raghumunukutla commented:
Breaking change was approved and suppression review was already performed on private preview PR and no more new suppressions are added. Here is the link for private pr: https://github.com/Azure/azure-rest-api-specs-pr/pull/15540 and below screenshot for reference. Removing ARMChangesRequested and putting in queue for review:

![image](https://github.com/Azure/azure-rest-api-specs/assets/55466801/38bb5d6d-ffe8-4b98-abc1-3487c468ffeb)

vimunuku committed:
Changes for updating type from number to integer as suggested by CLI team

rkmanda commented:
Please fix the [UniqueXmsEnumName](https://github.com/Azure/azure-openapi-validator/blob/main/docs/unique-xms-enum-name.md) . This was probably missed in the earlier PR but it might create problems when you go public with this API.

rkmanda commented:
Note to next reviewer: The AvoidAdditionalProperties failure can be ignored as the previous PR approved this usage.

rkmanda commented:
Please fix modelvalidation errors before going public. Not fixing these results in bad examples getting into your documentation.

rkmanda commented:
Also please fix [IntegerTypeMustHaveFormat](https://github.com/Azure/azure-openapi-validator/blob/main/docs/integer-type-must-have-format.md)

vimunuku committed:
Changes as per review comments

raghumunukutla commented:
> Please fix the [UniqueXmsEnumName](https://github.com/Azure/azure-openapi-validator/blob/main/docs/unique-xms-enum-name.md) . This was probably missed in the earlier PR but it might create problems when you go public with this API.

Addressed, provided the unique name

raghumunukutla commented:
> Also please fix [IntegerTypeMustHaveFormat](https://github.com/Azure/azure-openapi-validator/blob/main/docs/integer-type-must-have-format.md)

Addressed. Added the format.

raghumunukutla commented:
> Please fix modelvalidation errors before going public. Not fixing these results in bad examples getting into your documentation.

There were only errors related to avoid additional properties(below screenshot for reference)

![image](https://github.com/Azure/azure-rest-api-specs/assets/55466801/e7713744-adda-4af7-80d9-2d236487de52)

I have added the suppression rule in readme file as well but isn't sure why they are still showing up, please let me know if anything else needs to be done:

![image](https://github.com/Azure/azure-rest-api-specs/assets/55466801/5fbdd8ee-eb41-41c2-8f01-10a1e22f498a)

vimunuku committed:
Changes for adding suppression

weidongxu-microsoft commented:
/azp run

weidongxu-microsoft commented on specification/support/resource-manager/Microsoft.Support/preview/2023-06-01-preview/support.json:
```diff
@@ -0,0 +1,3431 @@
+    "ServiceClassificationAnswer": {
+      "description": "Service Classification result object.",
+      "type": "object",
+      "properties": {
+        "childService": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ClassificationService",
+          "description": "Child service."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/ClassificationService"
+        }
+      ]
```
Java codegen fails at here.

Please re-think whether you really want the `x-ms-client-flatten=true` on `childService`.

It is not recommended to flatten on non-`properties` property.

Also this `ServiceClassificationAnswer` already extends a `ClassificationService`. Having another `clientService` of same model and flatten into `ServiceClassificationAnswer` likely mess up the model.

vimunuku committed:
Based on recommendation from Java SDK team.

raghumunukutla commented:
/pr RequestMerge

vimunuku committed:
Changes in the operationid to have commandlets generated in a desired way for powershell/cli

raghumunukutla commented:
/pr RequestMerge
